configure.ac

```diff
@@ -5541,6 +5541,15 @@ AC_MSG_RESULT(python)
 AC_DEFINE(PY_SSL_DEFAULT_CIPHERS, 1)
 ])
 
+# Windows and HP-UX do not support strsignal(3)
+AC_MSG_CHECKING(for strsignal)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <string.h>]], [[
```

Comment:
Can't we add `strsignal` to the list at https://github.com/python/cpython/blob/7980d771c76d55aca55e1e619bada7c4707305b2/configure.ac#L3433 ?

Reply:
I have considered this, but it is  [nowhere documented](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Generic-Functions.html) that it will include `string.h`. I went for the safe bet.

Reply:
Many of the functions listed there require a header file to be included and none of them are documented in the autoconf documentation. For example, `alarm()` requires `unistd.h` and `Modules/signalmodules.c` already uses it:

https://github.com/python/cpython/blob/7980d771c76d55aca55e1e619bada7c4707305b2/Modules/signalmodule.c#L344

Does that check works on HP-UX? If it works, I think we can try adding `signalstr` to that list.

Currently, `configure.ac` is a mess, so it would nice if we can keep it shorter :)

Reply:
I will check that too.

Reply:
Just check on FreeBSD and HP-UX. It does isn't job in both `config.log` and `config.status` for both platforms. Will change accordingly.